modules/loadbalancer/main.tf

```diff
@@ -135,7 +136,7 @@ resource "azurerm_lb_rule" "lb_rules" {
   backend_address_pool_ids = [azurerm_lb_backend_address_pool.lb_backend.id]
 
   protocol                       = each.value.rule.protocol
-  backend_port                   = each.value.rule.port
+  backend_port                   = try(each.value.rule.backend_port, each.value.rule.port)
```

Comment:
Behaviour of functions `try` and `coalesce` is different, especially when `each.value.rule.backend_port` will have value `""`. Should we here use `coalesce` instead of `try` ?

Reply:
Actually the `each.value.rule.backend_port` might not exist at all in the object. How about the following `coalesce(try(each.value.rule.backend_port, null), each.value.rule.port)` ?

Reply:
Thanks!